Main.cs

```diff
@@ -0,0 +1,131 @@
+﻿var inventory = new Inventory();
+string name;
+decimal price; 
+int quantity, option;
+
+while (true)
+{
+    PrintMainMenu();
+    option = int.Parse(Console.ReadLine());
+
+    switch (option)
+    {
+        case 0:
+            return 0;
+
+        case 1:
+            Console.Write("Name of the product: ");
+            name = Console.ReadLine();
+            Console.Write("Price of the product: ");
+            price = decimal.Parse(Console.ReadLine());
+            Console.Write("Quantity of the product: ");
+            quantity = int.Parse(Console.ReadLine());
+            inventory.Add(name, price, quantity);
+            break;
+
+        case 2:
+            inventory.Print();
+            break;
+
+        case 3:
+            Console.Write("Name of a product to edit it: ");
+            name = Console.ReadLine();
+            Edit(name);
+            break;
+
+        case 4:
+            Console.Write("Name of a product to delete it: ");
+            name = Console.ReadLine();
+            inventory.Delete(name);
+            break;
+
+        case 5:
+            Console.Write("Name of a product to search for it: ");
+            name = Console.ReadLine();
+            var index = inventory.IndexOf(name);
+            if (index == -1)
+                Console.WriteLine($"{name} doesn't exist.");
+
+            else
+            {
+                Console.WriteLine();
+                inventory.Print(index);
+            }
+            break;
+
+        default:
+            Console.WriteLine("Please enter a valid option.");
+            break;
+    }
+
+    Console.Write("\nPress any key to continue.");
+    Console.ReadKey();
+    Console.Clear();
+}
+
+void Edit(string name)
```

Comment:
move this to inventory.cs, 
and fix the existing Edit method there.
remove the object newValue
u can use the code here to do the checks and the update at the same time (no need for 2 edit methods)